MainWindow.xaml.cs

```diff
@@ -81,5 +85,95 @@ private void btnLogin_Click(object sender, RoutedEventArgs e)
                 // Do something with the data
             }
         }
+        
+        private bool ValidateConfigXml(out string connStr)
+        {
+            connStr = null;
+            Dictionary<string,string> configs = null;
+
+            // Get the path to the AppData folder
+            string appDataPath = Globals.GetAppDataPath();
+            string filePath = Path.Combine(appDataPath, "config.xml");
+
+            // Validate config.xml exists
+            if (!File.Exists(filePath))
+            {
+                if (!TryConfigureDb(out configs)) return false;
+            }
+
+            // Read from config.xml and validate values are working
+            if (configs == null)
+            {
+                string server = "";
+                string port = "";
+                string username = "";
+                string dbName = "";
+
+                try
+                {
+                    Dictionary<string, string> fileConfigs = XmlHelper.GetContent(filePath);
+                    fileConfigs.TryGetValue("Server", out server);
+                    fileConfigs.TryGetValue("Port", out port);
+                    fileConfigs.TryGetValue("Username", out username);
+                    fileConfigs.TryGetValue("Password", out string password);
+                    fileConfigs.TryGetValue("DatabaseName", out dbName);
+
+                    string connTestStr = DbConfigDialog.BuildConnectionString(server, port, username, password, dbName);
+
+                    Mouse.OverrideCursor = Cursors.Wait;
+                    DbConfigDialog.DbConnect(connTestStr);
+                    Mouse.OverrideCursor = Cursors.Arrow;
+
+                    // Valid file config values
+                    configs = fileConfigs;
+                }
+                catch (Exception ex) when (ex is XmlException || ex is SecurityException)
+                {
+                    Mouse.OverrideCursor = Cursors.Arrow;
+                    MessageBox.Show(this, $"Unable to read config.xml\n {ex.Message}", Globals.AppName,
+                        MessageBoxButton.OK, MessageBoxImage.Error);
+
+                    if (!TryConfigureDb(out configs)) return false;
```

Comment:
The function that launches the configuration dialog should be extracted rather than embedded in this function.